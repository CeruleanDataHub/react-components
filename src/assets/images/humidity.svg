<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 89.84 90"><defs><style>.cls-1{fill:#04bbe7;}</style></defs><title>Humidity_icon</title><path class="cls-1" d="M79.34,17.74A43.21,43.21,0,0,0,48.59,5h0A43.5,43.5,0,0,0,17.82,79.26,43.17,43.17,0,0,0,48.57,92h0A43.29,43.29,0,0,0,64,89.2l-2.13-7.7A35.56,35.56,0,1,1,84,45.73c.07.92,7.22,11.3,7.22,11.3.24-1.19.42-2.38.56-3.58A43.55,43.55,0,0,0,79.34,17.74Z" transform="translate(-5.08 -5)"/><path class="cls-1" d="M80.34,48.5S65.76,68.56,65.76,79.67,72.29,95,80.34,95s14.58-4.23,14.58-15.33S80.34,48.5,80.34,48.5ZM76.89,91c-8.35-1.77-9.34-9.61-8.69-13.52a1.25,1.25,0,0,1,2.47.41c-.06.38-1.33,8.95,6.74,10.66a1.25,1.25,0,0,1-.26,2.48Z" transform="translate(-5.08 -5)"/><polygon class="cls-1" points="54.78 26.04 53.88 26.04 50.69 26.04 50.36 26.04 50.2 26.33 32.64 60.17 32.22 60.97 33.12 60.97 36.38 60.97 36.72 60.97 36.87 60.67 54.37 26.83 54.78 26.04"/><path class="cls-1" d="M59,51.27a7.35,7.35,0,1,0,7.33,7.9h0V58.08h0A7.36,7.36,0,0,0,59,51.27Zm2.46,9.8a3.48,3.48,0,1,1,1-2.45A3.45,3.45,0,0,1,61.42,61.07Z" transform="translate(-5.08 -5)"/><path class="cls-1" d="M45.52,38.93h0V37.84h0a7.35,7.35,0,1,0,0,1.09Zm-4.87,1.91a3.48,3.48,0,0,1-2.45,1,3.47,3.47,0,1,1,2.45-1Z" transform="translate(-5.08 -5)"/></svg>
