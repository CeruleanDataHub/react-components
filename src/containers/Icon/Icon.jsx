import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

const IconStyled = styled.span`
  font-size: 4rem;
  text-align: center;
`;

export const Icon = ({ name }) => (
  <IconStyled className={["lnr", `lnr-${name}`]} />
);

export const names = ["home", "users", "rocket", "briefcase", "cog", "eye"];
export const iconNameProps = PropTypes.oneOf(names);

Icon.propTypes = {
  name: iconNameProps,
};

Icon.defaultProps = {
  name: "",
};
