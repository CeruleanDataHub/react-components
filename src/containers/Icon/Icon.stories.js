import React from "react";
import styled from "styled-components";

import humidity from "../../assets/images/humidity.svg";
import { Icon, names } from "./Icon";

export default {
  title: "Icon",
  component: Icon
};

export const IconStory = () =>
  names.map(name => (
    <span style={{ display: "inline-block", margin: 20 }} key={name}>
      <Icon name={name} />
    </span>
  ));

const CustomIcon = styled.span`
  font-size: 1.5em;
  margin-right: 1.75rem;
  background-color: red;
`;

export const IconStyled = () => <Icon name="home" as={CustomIcon} />;

export const CustomIconStory = () => <Icon customIcon={humidity} />;
