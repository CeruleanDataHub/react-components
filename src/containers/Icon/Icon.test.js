import React from "react";
import renderer from "react-test-renderer";
import styled from "styled-components";

import humidity from "../../assets/images/humidity.svg";
import { Icon, names } from "./Icon";

describe("Icon", () => {
  describe("should render icon", () => {
    names.map(name =>
      it(name, () => {
        const component = renderer.create(<Icon name={name} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      })
    );
  });

  it("should render icon with custom styles", () => {
    const CustomIcon = styled.span`
      font-size: 1.5em;
      margin-right: 1.75rem;
    `;

    const component = renderer.create(<Icon name="home" as={CustomIcon} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render icon with custom icon image", () => {
    const component = renderer.create(<Icon customIcon={humidity} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
