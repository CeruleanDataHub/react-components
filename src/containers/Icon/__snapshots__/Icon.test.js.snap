// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render icon briefcase 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-briefcase"
/>
`;

exports[`Icon should render icon calendar 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-calendar"
/>
`;

exports[`Icon should render icon checkmark 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-checkmark"
/>
`;

exports[`Icon should render icon chevron-down 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-chevron-down"
/>
`;

exports[`Icon should render icon chevron-up 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-chevron-up"
/>
`;

exports[`Icon should render icon close 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-close"
/>
`;

exports[`Icon should render icon cog 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-cog"
/>
`;

exports[`Icon should render icon dashboard 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-dashboard"
/>
`;

exports[`Icon should render icon enter 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-enter"
/>
`;

exports[`Icon should render icon exit 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-exit"
/>
`;

exports[`Icon should render icon eye 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-eye"
/>
`;

exports[`Icon should render icon full-screen 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-full-screen"
/>
`;

exports[`Icon should render icon home 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-home"
/>
`;

exports[`Icon should render icon layers 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-layers"
/>
`;

exports[`Icon should render icon menu 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-menu"
/>
`;

exports[`Icon should render icon minus 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-minus"
/>
`;

exports[`Icon should render icon rocket 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-rocket"
/>
`;

exports[`Icon should render icon users 1`] = `
.c0 {
  text-align: center;
}

<span
  className="c0 lni lni-users"
/>
`;

exports[`Icon should render icon with custom styles 1`] = `
.c1 {
  text-align: center;
}

.c0 {
  font-size: 1.5em;
  margin-right: 1.75rem;
}

<span
  className="c0 c1 lni lni-home"
/>
`;
